src/cpp/session/modules/SessionLinter.cpp

```diff
@@ -291,16 +300,35 @@ Error getAllAvailableRSymbols(const FilePath& filePath,
    Error error;
    if (module_context::isRScriptInPackageBuildTarget(filePath) ||
        filePath.isWithin(projects::projectContext().directory().childPath("tests")))
+   {
+      DEBUG("- Package file:'" << filePath.absolutePath() << "'");
       error = getAvailableSymbolsForPackage(filePath, documentId, pSymbols);
+   }
    else
+   {
+      DEBUG("- Project file:'" << filePath.absolutePath() << "'");
       error = getAvailableSymbolsForProject(filePath, documentId, pSymbols);
+   }
    
    // If the file is within the 'tests/testthat' directory, then assume
    // 'testthat' will be available for those tests.
    if (filePath.isWithin(projects::projectContext().directory().childPath("tests/testthat")))
    {
       PackageSymbolRegistry& registry = packageSymbolRegistry();
-      registry.fillNamespaceExports("testthat", pSymbols);
+      registry.fillNamespaceObjects("testthat", pSymbols, false);
+   }
+   
+   // If the file is named 'server.R', 'ui.R' or 'app.R', we'll implicitly
```

Comment:
This is more rigorous detection logic: https://github.com/rstudio/rstudio/blob/master/src/cpp/session/modules/shiny/SessionShiny.cpp#L76

Is there a way we could re-use that? If it's too expensive to read the file contents you could also create another version that just inspects filenames.

Or, if you think "over-detecting" a reliance on shiny isn't a huge deal (it may not be) then we can leave as-is.